Services/IotHub/IotHubConnectionStringManager.cs

```diff
@@ -134,16 +146,35 @@ public void ValidateConnectionString(string connectionString)
             // if a key is provided, check if IoTHub is valid
             if (!match.Groups[CONNSTRING_REGEX_KEY].Value.IsNullOrWhiteSpace())
             {
+                RegistryManager registryManager = null;
                 try
                 {
-                    RegistryManager.CreateFromConnectionString(connectionString);
+                    registryManager = RegistryManager.CreateFromConnectionString(connectionString);
                 }
-                catch (IOException)
+                catch (Exception e)
                 {
                     string message = "Could not connect to IotHub with the connection" +
                                      "string provided. Check that the key is valid and " +
                                      "that the hub exists.";
-                    throw new IotHubConnectionException(message);
+                    throw new IotHubConnectionException(message, e);
+                }
+
+                try
+                {
+                    // check read permissions
+                    await registryManager.GetDevicesAsync(1, CancellationToken.None);
+
+                    // check write permissions
+                    var testDevice = new Device();
+                    testDevice = await registryManager.AddDeviceAsync(testDevice);
+                    await registryManager.RemoveDeviceAsync(testDevice);
+                }
+                catch (Exception e)
+                {
+                    string message = "Could not read or create devices with IotHub connection" +
+                                     "string provided. Check that the policy for the key allows " + 
+                                     "`Registry Read/Write` and `Service Connect` permissions.";
+                    throw new IotHubConnectionException(message, e);
```

Comment:
nit: rename the exception to something like "IoTHubConnectionTestException" (or "validation" or "connection string validation" etc.)